src/backend/mpi/write_buffer.hpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file
+ * @brief This file provides an interface for the ArgoDSM write buffer.
+ * @copyright Eta Scale AB. Licensed under the Eta Scale Open Source License. See the LICENSE file for details.
+ */
+
+#ifndef argo_write_buffer_hpp
+#define argo_write_buffer_hpp argo_write_buffer_hpp
+
+#include <deque>
+#include <iterator>
+#include <algorithm>
+
+/**
+ * @brief A write buffer in FIFO style with the capability to erase any
+ * element from the buffer while preserving ordering.
+ * @tparam T the type of the write buffer
+ */
+template<typename T>
+class WriteBuffer
+{
+	private:
+		/**
+		 * @brief This container holds cache indexes that should be written back
+		 */
+		std::deque<T> _buffer;
```

Comment:
good enough for now, but do you have an intuition about using another kind of structure here?

Reply:
Nothing in particular, open to suggestions and easy to change down the line.